### ISSUE TITLE
How to define a group matching categories?

### ISSUE THREAD

**Issue by novoid** [alphapapa/org-super-agenda#43]
Hi,

I'm trying to separate items from my "other items" section so that items from categories "foo" and "bar" are shown within their own group before the general "other" group with the rest starts.

I found all kind of criteria to define groups by tags, priority, deadlines, scheduled, effort, and so forth but nothing related to groups except auto-groups which is not what I want to use.

So: how am I able to define a group (before "other items") that contains all items from two specific categories that would otherwise end up in the "other items" group?

Thank you!

**Comment by alphapapa** [comment]
Hi Karl,

Please see this PR which I should merge soon: https://github.com/alphapapa/org-super-agenda/pull/38

**Comment by novoid** [comment]
Works like a charm! Thank you!

**Comment by alphapapa** [comment]
Great!  If you are ever interested in sending a screenshot example, please do, as I'm sure your advanced config would inspire many.  :)

**Comment by novoid** [comment]
If I may blur the text in-between the separators for privacy/security reasons, no problem.

However, my setup is not that different from most users I guess. Except the fact that it suits *my* requirements and that I did my setup somewhat more "complicated" as you have already seen: https://github.com/novoid/dot-emacs/blob/master/config.org#my-super-agenda

What I did with `:category` today was https://github.com/novoid/dot-emacs/blob/master/config.org#my-super-agenda-important-business-only and the switch between the two org-super-agenda configs using universal argument in https://github.com/novoid/dot-emacs/blob/master/config.org#my-org-agenda--my-map-a

If you think this is interesting, I'll do screenshots on Monday since my agenda is more interesting with business tasks ;-)

**Comment by alphapapa** [comment]
No problem, I'll leave it up to you whether you think your config is interesting enough to show off.  :)

BTW, I glanced at your config and there are a few things you could do more simply, e.g.:

```el
(:name "Important" :order 2 :and (:priority "A" :not (:tag "reward") :not (:tag "@BWG")))
```
Could be:
```el
(:name "Important" :order 2 :and (:priority "A" :not (:tag ("reward" "@BWG"))))
```
Because the `:tag` selector can take a list of multiple tags with implicit OR between them.  `:not` does the same, so you could also do something like:
```el
(:name "Space-related (non-moon-or-planet-related)"
       ;; Regexps match case-insensitively on the entire entry
       :and (:regexp ("space" "NASA")
                     ;; Boolean NOT also has implicit OR between selectors
                     :not (:regexp "moon" :tag "planet")))
```

:)

**Comment by alphapapa** [comment]
Also, I see that you exclude those two tags in nearly every group.  If you wanted to hide items with those tags completely (rather than having them show up in the "Other items" group at the end), you could put this as the first entry in `org-super-agenda-groups`:
```el
(:discard (:tag "reward" "@BWG"))
```

**Comment by novoid** [comment]
Good point, thank you. However, I want to see them in "Other items" in this case.

**Comment by alphapapa** [comment]
By the way, I happen to have this handy function laying around.  :)

```el
(defun org-agenda-sharpie ()
  "Censor the text of items in the agenda."
  (interactive)
  (let (regexp old-heading new-heading properties)
    ;; Save face properties of line in agenda to reapply to changed text
    (setq properties (text-properties-at (point)))

    ;; Go to source buffer
    (org-with-point-at (org-find-text-property-in-string 'org-marker
                                                         (buffer-substring (line-beginning-position)
                                                                           (line-end-position)))
      ;; Save old heading text and ask for new text
      (line-beginning-position)
      (unless (org-at-heading-p)
        ;; Not sure if necessary
        (org-back-to-heading))
      (setq old-heading (when (looking-at org-complex-heading-regexp)
                          (match-string 4))))
    (unless old-heading
      (error "Can't find heading.  How can this be?"))
    ;; Back to agenda buffer
    (setq new-heading (read-from-minibuffer "Overwrite visible heading with: "))
    (add-text-properties 0 (length new-heading) properties new-heading)
    ;; Replace agenda text
    (save-excursion
      (let ((inhibit-read-only t))
        (goto-char (line-beginning-position))
        (when (search-forward old-heading (line-end-position))
          (replace-match new-heading 'fixedcase 'literal))))))
```
